python/ray/data/_internal/datasource/hudi_datasource.py

```diff
@@ -0,0 +1,88 @@
+import logging
+from pathlib import PurePosixPath
+from typing import Dict, Iterator, List, Optional
+from urllib.parse import urljoin
+
+from ray.data._internal.util import _check_import
+from ray.data.block import BlockMetadata
+from ray.data.datasource.datasource import Datasource, ReadTask
+
+logger = logging.getLogger(__name__)
+
+
+class HudiDatasource(Datasource):
+    """Hudi datasource, for reading Apache Hudi table."""
+
+    def __init__(
+        self,
+        table_uri: str,
+        storage_options: Optional[Dict[str, str]] = None,
+    ):
+        _check_import(self, module="hudi", package="hudi-python")
+
+        self._table_uri = table_uri
+        self._storage_options = storage_options
+
+    def get_read_tasks(self, parallelism: int) -> List["ReadTask"]:
+        import pyarrow
+        from hudi import HudiTable
+
+        def _perform_read(
+            table_uri: str,
+            base_file_paths: List[str],
+            options: Dict[str, str],
+        ) -> Iterator["pyarrow.Table"]:
+            from hudi import HudiFileGroupReader
+
+            for p in base_file_paths:
+                fg_reader = HudiFileGroupReader(table_uri, options)
+                batch = fg_reader.read_file_slice_by_base_file_path(p)
+                yield pyarrow.Table.from_batches([batch])
+
+        hudi_table = HudiTable(self._table_uri, self._storage_options)
+
+        reader_options = {}
+        reader_options.update(hudi_table.storage_options())
+        reader_options.update(hudi_table.hudi_options())
+
+        schema = hudi_table.get_schema()
+        read_tasks = []
+        for file_slices in hudi_table.split_file_slices(parallelism):
+            if len(file_slices) <= 0:
```

Comment:
```suggestion
            if len(file_slices) == 0:
```

Reply:
BTW, how can split be empty?

Reply:
fixed. The table could be empty, the api returns 1 split of empty list

Reply:
Please elaborate that in a comment as it's not intuitive